q.js

```diff
@@ -1502,6 +1506,10 @@ Promise.prototype.keys = function () {
 Q.all = all;
 function all(promises) {
     return when(promises, function (promises) {
+        if (!isArray(arguments[0]) || arguments.length !== 1) {
+            throw Error("All must be passed an array of promises.");
```

Comment:
Would you mind replacing `All` with `all()` to make the error message more clear?